apps/ex_nvr/lib/ex_nvr/model/device.ex

```diff
@@ -158,6 +158,16 @@ defmodule ExNVR.Model.Device do
     end
   end
 
+  @spec base_url(t()) :: binary() | nil
+  def base_url(%__MODULE__{url: nil}), do: nil
+
+  def base_url(%__MODULE__{url: url}) do
+    case URI.parse(url) do
+      %{path: nil} -> url
+      %{path: path} -> String.replace(url, path, "")
+    end
```

Comment:
I think this is the same as writing
```elixir
url
|> URI.parse()
|> Map.put(:path, nil)
|> URI.to_string()
```
Let's also call the function `http_url/1`